packages/@lwc/shared/src/html-attributes.ts

```diff
@@ -72,50 +73,63 @@ export function isGlobalHtmlAttribute(attrName: string): boolean {
     return GLOBAL_ATTRIBUTE.has(attrName);
 }
 
-const HTML_ATTRIBUTES_TO_PROPERTY: Record<string, string> = {
-    accesskey: 'accessKey',
-    readonly: 'readOnly',
-    tabindex: 'tabIndex',
-    bgcolor: 'bgColor',
-    colspan: 'colSpan',
-    rowspan: 'rowSpan',
-    contenteditable: 'contentEditable',
-    crossorigin: 'crossOrigin',
-    datetime: 'dateTime',
-    formaction: 'formAction',
-    ismap: 'isMap',
-    maxlength: 'maxLength',
-    minlength: 'minLength',
-    novalidate: 'noValidate',
-    usemap: 'useMap',
-    for: 'htmlFor',
-};
+/**
+ * Map composed of properties to attributes not following the HTML property to attribute mapping
+ * convention.
+ */
+const NO_STANDARD_PROPERTY_ATTRIBUTE_MAPPING = new Map([
+    ['accessKey', 'accesskey'],
+    ['readOnly', 'readonly'],
+    ['tabIndex', 'tabindex'],
+    ['bgColor', 'bgcolor'],
+    ['colSpan', 'colspan'],
+    ['rowSpan', 'rowspan'],
+    ['contentEditable', 'contenteditable'],
+    ['crossOrigin', 'crossorigin'],
+    ['dateTime', 'datetime'],
+    ['formAction', 'formaction'],
+    ['isMap', 'ismap'],
+    ['maxLength', 'maxlength'],
+    ['minLength', 'minlength'],
+    ['noValidate', 'novalidate'],
+    ['useMap', 'usemap'],
+    ['htmlFor', 'for'],
+]);
 
-const HTML_PROPERTIES_TO_ATTRIBUTE: Record<string, string> = {};
-keys(HTML_ATTRIBUTES_TO_PROPERTY).forEach((attrName) => {
-    const propName = HTML_ATTRIBUTES_TO_PROPERTY[attrName];
-    HTML_PROPERTIES_TO_ATTRIBUTE[propName] = attrName;
-});
+/**
+ * Map associating previously transformed HTML property into HTML attribute.
+ */
+const CACHED_PROPERTY_ATTRIBUTE_MAPPING = new Map<string, string>();
 
-export function htmlAttributeToProperty(attrName: string): string {
-    if (attrName in AriaAttrNameToPropNameMap) {
-        return AriaAttrNameToPropNameMap[attrName];
+export function htmlPropertyToAttribute(propName: string): string {
+    const ariaAttributeName = AriaPropNameToAttrNameMap[propName];
+    if (!isUndefined(ariaAttributeName)) {
+        return ariaAttributeName;
     }
 
-    if (hasOwnProperty.call(HTML_ATTRIBUTES_TO_PROPERTY, attrName)) {
-        return HTML_ATTRIBUTES_TO_PROPERTY[attrName]!;
+    const specialAttributeName = NO_STANDARD_PROPERTY_ATTRIBUTE_MAPPING.get(propName);
+    if (!isUndefined(specialAttributeName)) {
+        return specialAttributeName;
     }
 
-    return attrName;
-}
-export function htmlPropertyToAttribute(propName: string): string {
-    if (propName in AriaPropNameToAttrNameMap) {
-        return AriaPropNameToAttrNameMap[propName];
+    const cachedAttributeName = CACHED_PROPERTY_ATTRIBUTE_MAPPING.get(propName);
+    if (!isUndefined(cachedAttributeName)) {
+        return cachedAttributeName;
     }
 
-    if (hasOwnProperty.call(HTML_PROPERTIES_TO_ATTRIBUTE, propName)) {
-        return HTML_PROPERTIES_TO_ATTRIBUTE[propName]!;
+    let attributeName = '';
+    for (let i = 0, len = propName.length; i < len; i++) {
```

Comment:
After running multiple benchmarks, this approach, it at least 2 to 3x faster than using the `String.prototype.replace` with a regular expression.